geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRemIntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void zRemDoesNotRemoveNonExistingMember() {
+    Map<String, Double> map = makeMemberScoreMap(INITIAL_MEMBER_COUNT);
+    addToSortedSet(map);
+
+    String nonExistingMember = "nonExisting";
+    long result = jedis.zrem(SORTED_SET_KEY, nonExistingMember);
+
+    assertThat(result).isEqualTo(0);
+  }
+
+  @Test
+  public void zRemCanRemoveAMemberInASortedSet() {
+    Map<String, Double> map = makeMemberScoreMap(INITIAL_MEMBER_COUNT);
+    Set<String> keys = map.keySet();
+    addToSortedSet(map);
+
+    String memberToRemove = baseName + 1;
+    Long removed = jedis.zrem(SORTED_SET_KEY, memberToRemove);
+    assertThat(removed).isEqualTo(1);
+
+    for (String member : keys) {
+      Double score = jedis.zscore(SORTED_SET_KEY, member);
+      if (member.equals(memberToRemove)) {
+        assertThat(score).isNull();
+      } else {
+        assertThat(score).isNotNull();
+      }
+    }
+    assertThat(jedis.exists(SORTED_SET_KEY)).isTrue();
+  }
+
+  @Test
+  public void zRemRemovesKeyIfRemoveAllMembersInASortedSet() {
+    Map<String, Double> map = makeMemberScoreMap(INITIAL_MEMBER_COUNT);
+    Set<String> keys = map.keySet();
+    addToSortedSet(map);
+
+    String[] membersToRemove = new String[keys.size()];
+    Long removed = jedis.zrem(SORTED_SET_KEY, keys.toArray(membersToRemove));
+    assertThat(removed).isEqualTo(keys.size());
+
+    for (String member : keys) {
+      Double score = jedis.zscore(SORTED_SET_KEY, member);
+      assertThat(score).isNull();
+    }
+    assertThat(jedis.exists(SORTED_SET_KEY)).isFalse();
+  }
+
+  @Test
+  public void zRemCanRemovesMembersConcurrentlyInASortedSet() {
+    int membersCount = 1000;
+    Map<String, Double> map = makeMemberScoreMap(membersCount);
+    addToSortedSet(map);
+
+    AtomicInteger totalRemoved = new AtomicInteger();
+    new ConcurrentLoopingThreads(2,
+        (i) -> doZRem(map, totalRemoved),
+        (i) -> doZRem1(membersCount, totalRemoved)).run();
+
+    assertThat(totalRemoved.get()).isEqualTo(membersCount);
+    assertThat(jedis.exists(SORTED_SET_KEY)).isFalse();
+  }
+
+  private void doZRem(Map<String, Double> map, AtomicInteger total) {
+    Set<String> keys = map.keySet();
+    for (String key : keys) {
+      long count = jedis.zrem(SORTED_SET_KEY, key);
+      total.addAndGet((int) count);
+    }
+  }
+
+  private void doZRem1(int memberCounts, AtomicInteger total) {
+    for (int i = 0; i < memberCounts; i++) {
+      long count = jedis.zrem(SORTED_SET_KEY, baseName + i);
+      total.addAndGet((int) count);
+    }
+  }
+
+  private void addToSortedSet(Map<String, Double> map) {
+    Set<String> keys = map.keySet();
+    Long count = 0L;
+
+    for (String member : keys) {
+      Double score = map.get(member);
+      Long res = jedis.zadd(SORTED_SET_KEY, score, member);
+      count += res;
+    }
+    assertThat(count).isEqualTo(keys.size());
+  }
```

Comment:
This method seems unnecessary since we can `zadd` all the keys/values in the map directly.